packages/react-reconciler/src/ReactFiberCommitWork.new.js

```diff
@@ -2007,6 +1900,22 @@ function commitWork(current: Fiber | null, finishedWork: Fiber): void {
             commitHydratedContainer(root.containerInfo);
           }
         }
+      } else {
+        // Persistent mode
+        const fiberRoot: FiberRoot = finishedWork.stateNode;
+        const containerInfo = fiberRoot.containerInfo;
+        const pendingChildren = fiberRoot.pendingChildren;
+        replaceContainerChildren(containerInfo, pendingChildren);
+      }
+      return;
+    }
+    case HostPortal: {
+      if (!supportsHydration) {
```

Comment:
I think you meant `!supportsMutation` here, not `!supportsHydration`.

That said, I'm also not sure whether we should use `!supportsMutation` or `supportsPersistence` as the condition. It's kind of confusing we have both but they're always mutually exclusive? I think maybe early on there was some renderer (noop?) that didn't support either? Anyway, I'm just not sure if it's better to write `!supportsMutation` or `supportsPersistence` or `!supportsMutation && supportsPersistence`.

Reply:
Yeah it should be `supportsMutation`. It happens to work because those two flags are always the same.

Re `supportsMutation` or `supportsPersistence`, yeah idk. We're not always consistent. It does seem like we check `supportsMutation` more often. Maybe should combine the flags.

Reply:
> always the same

usually the same, I should say. For RN and DOM, not the other renderers :)

Reply:
`supportsPersistence` feels like it fits better in this case

Reply:
Fixed: https://github.com/facebook/react/compare/6c608931b7c787fa5d6a6b66f8c7c83608e5a6e5..04d140c1138e3aa9dfdbf02a12faa7b49c28514e